src/app/modules/querybuilder/components/querybuilder-editor/save/save-dialog/save-dialog.component.ts

```diff
@@ -20,31 +24,21 @@ export class SaveDialogComponent implements OnInit {
   title = ''
   comment = ''
 
-  savedQueries: Array<{
-    query: Query
-    title: string
-    comment: string
-    date: number
-  }> = []
-
   ngOnInit(): void {
     this.query = this.queryProviderService.query()
-    this.savedQueries = this.queryProviderService.loadQueries()
-    if (this.savedQueries === undefined) {
-      this.savedQueries = []
-    }
   }
-
+  ngOnDestroy(): void {
+    this.subscriptionResult?.unsubscribe()
+  }
   doSave(): void {
-    this.savedQueries.push({
-      query: this.query,
-      title: this.title,
-      comment: this.comment,
-      date: Date.now(),
-    })
-    this.queryProviderService.saveQueries(this.savedQueries)
+    this.subscriptionResult?.unsubscribe()
+    this.subscriptionResult = this.backend
+      .saveQuery(this.query, this.title, this.comment)
```

Comment:
@thkoehler11  missing error handling.
if you save a query with the same name it is rejected by the backend with the following message:
HttpErrorResponse {headers: HttpHeaders, status: 409, statusText: 'Conflict', url: 'http://localhost:8090/api/v1/query-handler/stored-query', ok: false, …}

this should be reflected in the UI and the user should be presented with a message:

"Query with name <insert-name-here> already exists - please choose a different name"

please also note that the backend can throw an internal server error (status = 500) - this should be handled as well and displayed to the user:
"Query could not be saved - unknown error"